src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -335,6 +336,51 @@ class GraphFrame private(
    */
   def aggregateMessages: AggregateMessages = new AggregateMessages(this)
 
+  /**
+   * Filter the vertices according to Column expression, remove edges containing any dropped
+   * vertices.
+   * @group subgraph
+   */
+  def filterVertices(condExpr: Column): GraphFrame = {
+    val vv = vertices.filter(condExpr)
+    val ee = edges.join(vv, vv(ID) === edges(SRC), "left_semi")
+                  .join(vv, vv(ID) === edges(DST), "left_semi")
+    GraphFrame(vv, ee)
+  }
+
+  /**
+   * Filter the vertices according to String expression, remove edges containing any dropped
+   * vertices.
+   * @group subgraph
+   */
+  def filterVertices(condExpr: String): GraphFrame = filterVertices(expr(condExpr))
+
+  /** 
+   * Filter the edges according to Column expression, keep all vertices.
+   * @group subgraph
+   */
+  def filterEdges(condExpr: Column): GraphFrame = {
+    val vv = vertices
+    val ee = edges.filter(condExpr)
```

Comment:
if we are trying to maintain consistency when filtering vertices, when we filter edges, don't we want to make sure there is no "disconnected vertices"?

Reply:
why separately have dropIsolatedVertices?

Reply:
or instead it should be an optional param to choose which way to go?

Reply:
We have thought about the consistency for vertices. Instead of passing the optional parameter, we provide another function dropIsolatedVertices(). So users may use it if they want to maintain consistency. We think it's more clear compared to passing the optional flag. Besides, there is no cross- function optimization if we put these two functions together into one single function. 
```
filterEdges(cond).dropIsolatedVertices()
```

Reply:
I like how the current PR handles consistency.
@felixcheung With regards to select vs filter,
* Do you prefer one name to the other?  I like "filter" more actually since it is similar to Dataset.filter.
* The current API supports both Columns and String expressions.

Reply:
filter sounds good

Reply:
so I think we need to consider a few more cases?

`dropIsolatedVertices` is one, what about edges connecting to vertices not in `g.vertices`?

I think there's going to be a good set of "get graph consistent" functions or "check graph for consistency" functions

Reply:
Thanks for taking a look @felixcheung.

It's easy enough to add an API for `dropInvalidEdges()`, but what other "graph consistency" use cases did you have in mind? If there are a good number, maybe we should open a new issue for this because it seems like the "filter graph" and the "graph consistency" use cases seem separable to me.